subsys/usb/class/cdc_acm.c

```diff
@@ -328,36 +327,18 @@ static void cdc_acm_bulk_out(u8_t ep, enum usb_dc_ep_cb_status_code ep_status)
 	dev_data = CONTAINER_OF(common, struct cdc_acm_dev_data_t, common);
 
 	/* Check how many bytes were received */
-	usb_read(ep, NULL, 0, &bytes_to_read);
+	usb_read(ep, NULL, 0, &read);
 
-	buf_head = dev_data->rx_buf_head;
+	bytes_to_read = MIN(read, sizeof(dev_data->rx_buf));
 
-	/*
-	 * Quark SE USB controller is always storing data
-	 * in the FIFOs per 32-bit words.
-	 */
-	for (i = 0U; i < bytes_to_read; i += 4) {
-		usb_read(ep, tmp_buf, 4, NULL);
-
-		for (j = 0U; j < 4; j++) {
-			if (i + j == bytes_to_read) {
-				/* We read all the data */
-				break;
-			}
-
-			if (((buf_head + 1) % CDC_ACM_BUFFER_SIZE) ==
-			    dev_data->rx_buf_tail) {
-				/* FIFO full, discard data */
-				LOG_ERR("CDC buffer full!");
-			} else {
-				dev_data->rx_buf[buf_head] = tmp_buf[j];
-				buf_head = (buf_head + 1) % CDC_ACM_BUFFER_SIZE;
-			}
-		}
+	usb_read(ep, dev_data->rx_buf, bytes_to_read, &read);
+
+	if (!ring_buf_put(dev_data->rx_ringbuf, dev_data->rx_buf, read)) {
+		LOG_WRN("Rinf buffer full");
```

Comment:
New data will be dropped? Also we can not block here as it could be called from ISR (drivers without worker or own thread). I would change it to LOG_ERR.

Reply:
@jfischer-phytec-iot why do you think it can block? The new data would be dropped the same way it was done before, functionality should not change.

Reply:
Changed to LOG_ERR()